### ISSUE TITLE
`time(...)` does not support the microsecond `%f` formatter

### ISSUE THREAD

**Issue by Xyene** [bpftrace/bpftrace#3568]
The docs say this about `time(...)`:

> bpftrace uses the `strftime(3)` function for formatting time and supports the same format specifiers.

However, it doesn't seem to format `%f` correctly, whereas `strftime(...)` does.

For example:

```c
$ bpftrace -e 'tracepoint:exceptions:page_fault_user { time("%H:%M%S.%f\n") }'
04:0742.%f
04:0742.%f
04:0742.%f
...
```

While using `strftime` directly works:

```c
$ bpftrace -e 'tracepoint:exceptions:page_fault_user { printf("%s\n", strftime("%H:%M%S.%f", nsecs)) }'
04:0950.872453
04:0950.872456
04:0950.872458
...
```

This is on master, built a few weeks ago.

<details>
<summary><code>bpftrace --info</code></summary>

```
System
  OS: [redacted]
  Arch: x86_64

Build
  version: v0.21.0-167-g2793-dirty
  LLVM: 17.0.6
  bfd: yes
  liblldb (DWARF support): no
  libsystemd (systemd notify support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no
  get_boot_ns: yes
  dpath: yes
  skboutput: yes
  get_tai_ns: yes
  get_func_ip: yes
  jiffies64: yes
  for_each_map_elem: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: yes
  module btf: yes
  map batch: yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes
  ringbuf: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
  kprobe_multi: no
  uprobe_multi: no
  raw_tp_special: yes
  iter: yes

```

</details>

**Comment by amiremohamadi** [comment]
according to the [manual](https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#time):
> bpftrace uses the strftime(3) function for formatting time and supports the same format specifiers.

the `%f` format string is a specific extension for bpftrace's [strftime](https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#strftime) built-in function.

**Comment by Xyene** [comment]
Aha, I hadn't realized `%f` was an extension.

I haven't dug into the code, but is `time` handled materially differently from `strftime`, or could it be similarly extended?

**Comment by amiremohamadi** [comment]
higher resolution is achievable but not with the same approach -- as it uses [`time`](https://man7.org/linux/man-pages/man7/time.7.html) function which represents seconds since the UNIX epoch.

though i'm not sure if there's any real use cases for having a higher resolution for our `time` built-in.
cc: @danobi

**Comment by danobi** [comment]
Are there use cases for time() that strftime() cannot address? When we added strftime() we probably should have deprecated time(), as it is far less flexible and actually less accurate.

**Comment by danobi** [comment]
Fair enough. FWIW you could also do `print(strftime(...))` to shorten it a little bit. But long term we should probably focus our efforts on a single helper. So I'm going to close this out as a wont-fix. We appreciate the report, though.
